docs/guide/state.md

```diff
@@ -0,0 +1,569 @@
+[alGetDoublev](#algetdoublev), [alGetFloat](#algetfloat),
+[alGetFloatv](#algetfloatv), [alGetInteger](#algetinteger),
+[alGetIntegerv](#algetintegerv)
+
+#### alGetDouble
+##### Description
+This function returns a double precision floating point OpenAL state.
+
+```cpp
+ALdouble alGetDouble(
+    ALenum param
+);
+```
+
+##### Parameters
+* param - the state to be queried:
+    + `AL_DOPPLER_FACTOR`
+	+ `AL_SPEED_OF_SOUND`
+	+ `AL_DISTANCE_MODEL`
+
+##### Possible Error States
+| State                | Description |
+| -------------------- | ----------- |
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+| AL_INVALID_OPERATION | There is no current context. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+The double value described by `param` will be returned.
+
+##### See Also
+[alGetBoolean](#algetboolean), [alGetBooleanv](#algetbooleanv),
+[alGetDoublev](#algetdoublev), [alGetFloat](#algetfloat),
+[alGetFloatv](#algetfloatv), [alGetInteger](#algetinteger),
+[alGetIntegerv](#algetintegerv)
+
+#### alGetFloat
+##### Description
+This function returns a floating point OpenAL state.
+
+```cpp
+ALfloat alGetFloat(
+    ALenum param
+);
+```
+
+##### Parameters
+* param - the state to be queried:
+    + `AL_DOPPLER_FACTOR`
+	+ `AL_SPEED_OF_SOUND`
+	+ `AL_DISTANCE_MODEL`
+
+##### Possible Error States
+| State                | Description |
+| -------------------- | ----------- |
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+| AL_INVALID_OPERATION | There is no current context. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+The floating point state described by `param` will be returned.
+
+##### See Also
+[alGetBoolean](#algetboolean), [alGetBooleanv](#algetbooleanv),
+[alGetDouble](#algetdouble), [alGetDoublev](#algetdoublev),
+[alGetFloatv](#algetfloatv), [alGetInteger](#algetinteger),
+[alGetIntegerv](#algetintegerv)
+
+#### alGetInteger
+##### Description
+This function returns an integer OpenAL state.
+
+```cpp
+ALint alGetInteger(
+    ALenum param
+);
+```
+
+##### Parameters
+* param - the state to be queried:
+    + `AL_DOPPLER_FACTOR`
+	+ `AL_SPEED_OF_SOUND`
+	+ `AL_DISTANCE_MODEL`
+
+##### Possible Error States
+| State                | Description |
+| -------------------- | ----------- |
+| AL_INVALID_ENUM      | The specified parameter is not valid. |
+| AL_INVALID_OPERATION | There is no current context. |
+
+##### Version Requirements
+OpenAL 1.0 or higher
+
+##### Remarks
+The integer state described by `param` will be returned.
+
+##### See Also
+[alGetBoolean](#algetboolean), [alGetBooleanv](#algetbooleanv),
+[alGetDouble](#algetdouble), [alGetDoublev](#algetdoublev),
+[alGetFloat](#algetfloat), [alGetFloatv](#algetfloatv),
+[alGetIntegerv](#algetintegerv)
+
+#### alGetBooleanv
+##### Description
+This function returns an integer OpenAL state.
+
+```cpp
+void alGetBooleanv(
+    ALenum param,
+	ALboolean* data

```

Comment:
Not sure if this was intentional, but a number of functions seem to have extra indentation on some parameters.